mm2src/lp_ordermatch.rs

```diff
@@ -3285,6 +3341,130 @@ pub async fn set_price(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, Strin
+
+    // Validate and Add new_price to update_msg if new_price is found in the request
+    if let Some(new_price) = req.new_price.clone() {
+        try_s!(validate_price(new_price.clone()));
+        update_msg = update_msg.with_new_price(new_price.into());
+    }
+
+    // Add min_volume to update_msg if min_volume is found in the request
+    if let Some(min_volume) = req.min_volume.clone() {
+        update_msg = update_msg.with_new_min_volume(min_volume.into());
+    };
+
+    let min_base_amount = base_coin.min_trading_vol();
+    let min_rel_amount = rel_coin.min_trading_vol();
+
+    // Validate and Calculate Minimum Volume if min_volume is not in the request
+    let actual_min_base_vol = try_s!(validate_and_get_min_vol(
+        min_base_amount.clone(),
+        min_rel_amount.clone(),
+        req.min_volume.clone(),
+        req.new_price.clone().unwrap_or_else(|| my_order.clone().price)
+    ));
+
+    // Calculate order volume and add to update_msg if new_volume is found in the request
+    let volume = if req.max.unwrap_or(false) {
+        let max_volume = try_s!(get_max_volume(&ctx, &base_coin, &rel_coin).await);
+        update_msg = update_msg.with_new_max_volume(max_volume.clone().into());
+        max_volume
+    } else if Option::is_some(&req.new_volume) {
+        try_s!(
+            check_balance_for_maker_swap(
+                &ctx,
+                &base_coin,
+                &rel_coin,
+                req.new_volume.clone().unwrap(),
+                None,
+                None,
+                FeeApproxStage::OrderIssue
+            )
+            .await
+        );
+        update_msg = update_msg.with_new_max_volume(req.new_volume.clone().unwrap().into());
+        req.new_volume.unwrap()
+    } else {
+        my_order.max_base_vol.clone()
+    };
+
+    // Validate Order Volume
+    try_s!(validate_max_vol(
+        min_base_amount.clone(),
+        min_rel_amount.clone(),
+        volume.clone(),
+        req.min_volume.clone(),
+        req.new_price.clone().unwrap_or_else(|| my_order.clone().price)
+    ));
+
+    let conf_settings = OrderConfirmationsSettings {
+        base_confs: req
+            .base_confs
+            .unwrap_or_else(|| my_order.conf_settings.unwrap().base_confs),
+        base_nota: req
+            .base_nota
+            .unwrap_or_else(|| my_order.conf_settings.unwrap().base_nota),
+        rel_confs: req
+            .rel_confs
+            .unwrap_or_else(|| my_order.conf_settings.unwrap().rel_confs),
+        rel_nota: req.rel_nota.unwrap_or_else(|| my_order.conf_settings.unwrap().rel_nota),
+    };
+
+    update_msg = update_msg.with_new_conf_settings(conf_settings);
+
+    let updated_order = MakerOrder {
+        base: base.to_string(),
+        rel: rel.to_string(),
+        created_at: my_order.created_at,
+        updated_at: now_ms(),
+        max_base_vol: volume,
+        min_base_vol: actual_min_base_vol,
```

Comment:
This can reset the min_base_vol to a lower amount even if it's not set in the request that will be unexpected by the user.
E.g. if the initial order's `min_base_vol` is 1 but coin min_trading_vol is 0.0001. It can even lead to a potential financial loss as `min_base_vol` is used to set a reasonable trade amount for high fee coins (BTC, ETH, etc https://github.com/KomodoPlatform/atomicDEX-API/issues/473#issue-469539625)